control-plane/agents/src/bin/core/tests/rebuild/mod.rs

```diff
@@ -0,0 +1,260 @@
+        .with_reconcile_period(reconcile_period, reconcile_period)
+        .with_cache_period("1s")
+        .with_tmpfs_pool(pool_size)
+        .build()
+        .await
+        .unwrap()
+}
+
+async fn wait_till_pool_online(
+    cluster: &Cluster,
+    nodeidx: u32,
+    poolidx: u32,
+    timeout: Duration,
+) -> Option<Pool> {
+    let client = cluster.rest_v00();
+    let pool_api = client.pools_api();
+    let poolid = cluster.pool(nodeidx, poolidx);
+    let start = std::time::Instant::now();
+
+    loop {
+        let pool = pool_api.get_pool(poolid.as_str()).await.unwrap();
+        tracing::info!("Waiting for pool to be Online - {:?}", pool);
+        if pool.state.is_some() && (pool.state.clone().unwrap().status == PoolStatus::Online) {
+            tracing::info!("Pool is Online now");
+            return Some(pool);
+        }
+
+        if std::time::Instant::now() > (start + timeout)
+            && (pool.state.is_none() || pool.state.clone().unwrap().status != PoolStatus::Online)
+        {
+            return None;
+        }
+        tokio::time::sleep(Duration::from_secs(1)).await;
+    }
+}
+
+// This test
+// 1. creates a volume with 3 replicas, faults one of them by restarting container.
+// 2. Expects child's replica to be online again within 15sec period.
+// 3. Attempts to online the child.
+// 4. Child should get successfully onlined and partial rebuild happen.
+// 5. If online call fails, full rebuild of child happens.
+#[tokio::test]
+async fn replica_wait_and_partial_rebuild() {
+    let cluster = build_cluster(4, 52428800, 0).await; // 50MiB pool size
+
+    let vol_target = cluster.node(0).to_string();
+    let api_client = cluster.rest_v00();
+    let volume_api = api_client.volumes_api();
+
+    let body = CreateVolumeBody::new(VolumePolicy::new(true), 3, 10485760u64, false);
+    let volid = VolumeId::new();
+    let volume = volume_api.put_volume(&volid, body).await.unwrap();
+    let volume = volume_api
+        .put_volume_target(
+            &volume.spec.uuid,
+            PublishVolumeBody::new_all(
+                HashMap::new(),
+                None,
+                vol_target.clone().to_string(),
+                models::VolumeShareProtocol::Nvmf,
+                None,
+                cluster.csi_node(0),
+            ),
+        )
+        .await
+        .unwrap();
+
+    let volume_state = volume.state;
+    assert!(volume_state.status == VolumeStatus::Online.into());
+    let nexus = volume_state.target.unwrap();
+    assert_eq!(nexus.children.len(), 3);
+    nexus
+        .children
+        .iter()
+        .for_each(|c| assert!(c.state == ChildState::Online.into()));
+
+    // Check where the replicas are, apart from vol target node.
+    let replicas = api_client.replicas_api().get_replicas().await.unwrap();
+    tracing::info!({?replicas}, "Available replicas - ");
+    let testrep = replicas.iter().find(|r| r.node != vol_target).unwrap();
```

Comment:
what is testrep?

Reply:
testrep indicates the _test replica_ that is picked up for this test.